modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -641,6 +641,17 @@ void ONNXImporter::populateNet(Net dstNet)
                     {
                         layerParams.type = "Scale";
                         layerParams.set("bias_term", true);
+                        int axis = 1;
+                        for (int i = 0; i < graph_proto.initializer_size(); i++)
```

Comment:
Why we need to iterate over all initializers? Is there existing map or something?

Reply:
I only know one way to get the original blob sizes: `tensor_proto.dims_size ()`, because for `tensor_proto.dims = {2, 1}` and `tensor_proto.dims = {2}` we get cv :: Mat with the same sizes: {2, 1}

Reply:
Does `getBlob` preserve dimensions?

Reply:
For `tensor_proto.dims_size() == 1 or tensor_proto.dims_size() == 2` `getBlob` returns 2D Mat

Reply:
Can you try to change
https://github.com/opencv/opencv/blob/06bff34a6b4dedb6f187b6bb98e914d6d6701b39/modules/dnn/src/onnx/onnx_graph_simplifier.cpp#L528-L529
to 
```cpp
blob.dims = std::max(1, tensor_proto.dims_size());
```
?

Reply:
I tried to do this, but in this case we get errors when we try to push_back 1D Mat in `layerParams`

Reply:
Well, Ok. Let's keep as is.